Packs/SymantecEDR/Integrations/SymantecEDR/SymantecEDR.py

```diff
@@ -0,0 +1,2734 @@
+    410: '410 Gone - Access to the target resource is no longer available at the origin server',
+    500: '500 Internal Server Error - please try again after some time.',
+    502: '502 Bad Gateway - Could not connect to the origin server',
+    503: '503 Service Unavailable'
+}
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """
+    This Client implements API calls, and does not contain any XSOAR logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this implementation, no special attributes defined
+    """
+    def __init__(self, base_url: str, verify: bool, proxy: bool, client_id: str, client_secret: str,
+                 first_fetch: str = '3 days', fetch_limit: Optional[int] = 50, is_incident_event: bool = False,
+                 is_fetch_comment: bool = False, fetch_status: list = None, fetch_priority: list = None,
+                 token: Optional[str] = None):
+
+        super().__init__(
+            base_url=base_url,
+            verify=verify,
+            proxy=proxy
+        )
+
+        self.client_key = client_id
+        self.secret_key = client_secret
+        self.first_fetch = first_fetch
+        self.fetch_limit = fetch_limit
+        self.is_incident_event = is_incident_event
+        self.is_fetch_comment = is_fetch_comment
+        self.fetch_status = fetch_status
+        self.fetch_priority = fetch_priority
+        self.access_token = token
+
+    @property
+    def headers(self) -> dict:
+        """
+        Client headers method
+        Returns:
+            self.headers
+        """
+        if self.access_token is None:  # for logging in, before self.access_token is set
+            return {'Content-Type': 'application/json'}
+
+        return {'Authorization': f'Bearer {self.access_token}', 'Content-Type': 'application/json'}
+
+    @staticmethod
+    def get_access_token_from_context(global_context: dict[str, Any]) -> str | None:
+        """
+        Symantec EDR on-premise get previous access token from global integration context
+        Args:
+            global_context(dict): Integration Context data
+        Returns:
+            return token or None
+        """
+        if save_timestamp := global_context.get('access_token_timestamp'):
+            now_timestamp = int(time.time())
+            if token := global_context.get('access_token'):
+                time_diff = int(now_timestamp - save_timestamp)
+                if time_diff <= SESSION_TIMEOUT_SEC:
+                    return token
+                else:
+                    LOG('Access token expired')
+        return None
+
+    def get_access_token_or_login(self) -> None:
+        """
+        Generate Access token
+        """
+        global_context = demisto.getIntegrationContext()
+        if last_access_token := self.get_access_token_from_context(global_context):
```

Comment:
please move `demisto.getIntegrationContext()` into `get_access_token_from_context`, instead of passing it as argument

Reply:
Corrected. Thank you

Reply:
Changes done.